tai_unpack.c

```diff
@@ -0,0 +1,16 @@
+/* Public domain. */
+
+#include "tai.h"
+
+void tai_unpack(const char *s,struct tai *t)
```

Comment:
I think the interface of this function is unfortunate. When called it is called with a struct tai, so the pointer needs to be casted. But internally this casts again. So the interface should either take a unsigned char* directly, or just take void* and cast it to an unsigned char* internally that can be used without continuous casting. I'd slightly prefer the latter.